README.md

```diff
@@ -231,3 +231,218 @@ var_dump($taichungService->getAction1()->do()->getMeaningData());
 var_dump($taichungService->getAction2()->do()->getMeaningData());
 ```
+
+### HTTP JSON RPC Single Request
+
+In the `Action` object, you can pass the desired `method`, `params`, and `id` for the `RPC` connection by using the `setRpcQuery()` method.
+
+Once the `setRpcQuery()` method is enabled, the HTTP verb of the `Action` will automatically change to `POST`.
+
+By setting the `doneHandler`, you can define the execution logic when the connection is successful, and you can temporarily store the required data in the `Action` instance using `setMeaningData`.
+
+If the `RPC` response is parsed correctly, you can handle your `RPC` response using the `doneHandler`.
+
+You can use `getRpcResponse()` to obtain the successful RPC response entity, `getRpcResult()` to get the RPC response data (i.e., result), and `getId()` to retrieve the RPC response ID. All of these methods return their results as arrays.[Reference to JSON-RPC format](https://www.jsonrpc.org/specification "Reference to JSON-RPC format")。
+
+```php
+require './vendor/autoload.php';
+use SDPMlab\Anser\Service\Action;
+use Psr\Http\Message\ResponseInterface;
+use SDPMlab\Anser\Service\ServiceList;
+
+$id = 1;
+$action = (new Action(
+    'http://myRpcServer.com',
+    "POST",
+    "/"
+))
+->setRpcQuery("/myRpcMethod", [1,2], $id)
+->doneHandler(static function(
+    ResponseInterface $response,
+    Action $runtimeAction
+) {
+    $rpcResponse = $action->getRpcResponse();
+    $result = $action->getRpcResult()[0];
+    $id     = $action->getId()[0];
+    $runtimeAction->setMeaningData([
+        "id" => $id,
+        "result" => $result
+    ]);
+});
+
+$data = $action->do()->getMeaningData();
+var_dump($data);
+```
+
+### HTTP JSON RPC Batch request
+
+Similar to the JSON RPC single request approach, you can use the `setBatchRpcQuery()` method in the `Action` object to pass the desired `method`, `params`, and `id` as an array to perform batch RPC connections.
+
+Once the `setBatchRpcQuery()` method is enabled, the HTTP verb of the `Action` will automatically change to `POST`.
+
+If all batch `RPC` responses are successfully parsed, you can handle your `RPC` responses using the `doneHandler`.
+
+```php
+require './vendor/autoload.php';
+use SDPMlab\Anser\Service\Action;
+use Psr\Http\Message\ResponseInterface;
+use SDPMlab\Anser\Service\ServiceList;
+
+$id = 1;
+$action = (new Action(
+    'http://myRpcServer.com',
+    "POST",
+    "/"
+))
+->setBatchRpcQuery([
+    ["/myRpcMethod", [1,2], $id],
+    ["/myRpcMethod", [1,2], $id],
+    ["/myRpcMethod", [1,2], $id]
+]); 
+->doneHandler(static function(
+    ResponseInterface $response,
+    Action $runtimeAction
+) {
+    $returnData = [];
+    $rpcResponse = $action->getRpcResponse();
+    $resultArray = $action->getRpcResult();
+    $idArray     = $action->getId();
+    for ($i = 0; $i<3; $i++ ){
+        $returnData[] = [
+            "id" => $idArray[$i],
+            "result" => $resultArray[$i],
+        ]
+    }
```

Comment:
I think the return result of `$action->getRpcResponse();` could be a Key/Value array with `QueryID` as the key.

Reply:
Done.